### ISSUE TITLE
[🐛 Bug]: @wdio/cucumber-framework depends on mockery which is vulnerable

### ISSUE THREAD

**Issue by remdash** [webdriverio/webdriverio#10818]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md#reporting-new-issues).

### WebdriverIO Version

7.19.7 (but the issue also exists with the latest version 8.13.13) 

### Node.js Version

16.17.0

### Mode

WDIO Testrunner

### What happened?

Since the vulnerability [CVE-2022-37614](https://github.com/advisories/GHSA-gmwp-3pwc-3j3g) for `mockery`  has been reviewed on the 29th of July 2023, a package that uses `@wdio/cucumber-framework` as a dependency will now fails the pnpm audit.

Running ` pnpm audit --audit-level critical` for such a package  will now yield an error as `mockery` is a dependency of `@wdio/cucumber-framework`. 

Additionaly, as mockery hasn't been updated since 2017 and has no fixed version, it is not possible to use a forced resolution of a higher version of it.

### What is your expected behavior?

`@wdio/cucumber-framework` uses up-to-date dependencies that have no critical vulnerabilities.

### How to reproduce the bug.

To reproduce the bug, just add `@wdio/cucumber-framewok` to the package.json file of a new project and install with pnpm:
`pnpm add @wdio/cucumber-framework` 
`pnpm i`
Then run the audit tool:
`pnpm audit --audit-level critical` 
and you should see the error.

### Relevant log output

```typescript
pnpm audit --audit-level critical
┌─────────────────────┬───────────────────────────────────────────────────┐
│ critical            │ mockery is vulnerable to prototype pollution      │
├─────────────────────┼───────────────────────────────────────────────────┤
│ Package             │ mockery                                           │
├─────────────────────┼───────────────────────────────────────────────────┤
│ Vulnerable versions │ <=2.1.0                                           │
├─────────────────────┼───────────────────────────────────────────────────┤
│ Patched versions    │ <0.0.0                                            │
├─────────────────────┼───────────────────────────────────────────────────┤
│ More info           │ https://github.com/advisories/GHSA-gmwp-3pwc-3j3g │
└─────────────────────┴───────────────────────────────────────────────────┘
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

**Comment by christian-bromann** [comment]
Thanks for reporting!

Given that this package hasn't received any new releases for 6 years it makes sense to replace it with something more modern.

We greatly appreciate any contributions that help resolve the bug. While we understand that active contributors have their own priorities, we kindly request your assistance if you rely on this bug being fixed. We encourage you to take a look at our [contribution guidelines](https://webdriver.io/docs/contribute) or join our friendly [Discord](http://discord.webdriver.io/) development server, where you can ask any questions you may have. Thank you for your support, and cheers!

**Comment by DanielFreemanTester** [comment]
I'm also seeing this issue with node v18.16.0, @wdio/cli v8.3.3 and @wdio/cucumber-framework v8.3.0 (also the latest v8.14.0).

Any chance somebody could fix this bug please? I would imagine it will affect a significant number of WebdriverIO users.

**Comment by christian-bromann** [comment]
> Any chance somebody could fix this bug please?

Please get involved in the project if you have interest in a fix.

**Comment by DanielFreemanTester** [comment]
Thanks @christian-bromann , I'm not a developer, I'm definitely a tester but I will pass this issue along to one of our developers on  our project on the slim chance that they would have the time + inclination to do this. I appreciate open source is all about contribution but I'm not in a position to do this myself, sorry.

**Comment by remdash** [comment]
> Thanks for reporting!
> 
> Given that this package hasn't received any new releases for 6 years it makes sense to replace it with something more modern.
> 
> We greatly appreciate any contributions that help resolve the bug. While we understand that active contributors have their own priorities, we kindly request your assistance if you rely on this bug being fixed. We encourage you to take a look at our [contribution guidelines](https://webdriver.io/docs/contribute) or join our friendly [Discord](http://discord.webdriver.io/) development server, where you can ask any questions you may have. Thank you for your support, and cheers!

Thanks, I'll have a look.

**Comment by remdash** [comment]
After investigating, mockery  seems to be used only in the Cucumber adapter to make sure to use the cucumber version shipped in with wdio instead of another one globally installed for example.

So, I'm not sure of where to go from there as I can see multiple options.

The first option is to check if is it still needed to mock the `@cucumber/cucumber` import or is there another way to make sure that the correct version is used (through some peerDependencies in the package.json for example).

If it is still needed to maintain the structure of the code, the next option is to look for a more maintained alternative.  After looking at some alternatives libraries, I'm not sure of which one may be suitable. `Proxyquire` has last been modified 4 years ago . `sinon` does have some limitations in its stub that prevent to replace a full import.
 `rewire` or `rewiremock` may be more appropriated as they would allow to mock the import and seems to be more recently maintained.
 
Given that I lack context on this repository, I would need some advice at this point before moving on.

**Comment by christian-bromann** [comment]
> `rewire` or `rewiremock` may be more appropriated as they would allow to mock the import and seems to be more recently maintained.

Seems like this would be a good option.

**Comment by remdash** [comment]
Thanks for the answer, I won't have the time for now to work on this further than that but I hope that the investigation will still be useful to update that part of the code base.

**Comment by DanielFreemanTester** [comment]
Thanks very much @remdash for the time you have been able to spend on this - much appreciated.

**Comment by christian-bromann** [comment]
Fixed by #11010 .. thanks @tamil777selvan
